charts/super-agent/charts/super-agent-deployment/templates/_helpers.tpl

```diff
@@ -2,19 +2,80 @@
 Return the name of the configMap holding the Super Agent's config. Defaults to release's fill name suffiexed with "-config"
 */ -}}
 {{- define "newrelic-super-agent.config.name" -}}
-{{- (include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" "local-data" "suffix" "superagent-config" )) -}}
+{{- include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" "local-data" "suffix" "superagent-config" ) -}}
 {{- end -}}
 
+
+
+{{- /*
+Test that the value of `.Values.config.subAgents` exists and its valid. If empty, returns the default.
+*/ -}}
+{{- define "newrelic-super-agent.config.agents.yaml" -}}
+{{- if (.Values.config).subAgents -}}
+{{- $agents := dict -}}
+{{- range $subAgentName, $subAgentConfig := (.Values.config).subAgents -}}
+  {{- if not ($subAgentConfig).type -}}
+    {{- fail (printf "Agent %s does not have agent type" $subAgentName) -}}
+  {{- end -}}
+  {{- $agents = mustMerge $agents (dict $subAgentName $subAgentConfig) -}}
+{{- end -}}
+{{- $agents | toYaml -}}
+{{- else -}}
+{{- /* Default agents for Kubernetes */ -}}
+open-telemetry:
```

Comment:
out of curiosity , this default is on the helper to avoid the merge when custom ones are added? if so , isn't that against the expected behaviour of helm values ?

Reply:
In Helm, objects are merged and lists are overridden. In that sense you are right.

But if we do not allow to override it this way, then the `open-telemetry` subagent is enabled always and that is no what we want.

This is a common behavior while creating complex defaults and we already do this with the `common-library` in [many](https://github.com/newrelic/helm-charts/blob/ddd7eb10b7530d3282f5469ddd3321a60192e524/library/common-library/templates/_nodeselector.tpl), [many](https://github.com/newrelic/helm-charts/blob/ddd7eb10b7530d3282f5469ddd3321a60192e524/library/common-library/templates/_low-data-mode.tpl), [many](https://github.com/newrelic/nri-kubernetes/blob/4545f562851f739b71ac86f9dc3d49cfd3a85b06/charts/newrelic-infrastructure/templates/kubelet/_security_context_helper.tpl#L25-L29) cases.